crash - BuildIEMobilityRestrictionList

### Description

I don't have logs, only crash info:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x9e7054]

goroutine 69 [running]:
github.com/ellanetworks/core/internal/amf/ngap/message.BuildIEMobilityRestrictionList(0x40003b6488)
github.com/ellanetworks/core/internal/amf/ngap/message/forward_ie.go:151 +0x214
github.com/ellanetworks/core/internal/amf/ngap/message.BuildInitialContextSetupRequest({0x16870a8, 0x40004872c0}, 0x40003b6488, {0x1251d75, 0xb}, {0x40004d9100, 0x34, 0x40}, 0x400050e7f8, 0x0, ...)
github.com/ellanetworks/core/internal/amf/ngap/message/build.go:1016 +0x160c
github.com/ellanetworks/core/internal/amf/ngap/message.SendInitialContextSetupRequest({0x16870a8?, 0x40004872c0?}, 0x40003b6488, {0x1251d75, 0xb}, {0x40004d9100?, 0x0?, 0x0?}, 0x0?, 0x0?, ...)
github.com/ellanetworks/core/internal/amf/ngap/message/send.go:350 +0x78
github.com/ellanetworks/core/internal/amf/gmm/message.SendRegistrationAccept({0x16870a8, 0x40004872c0}, 0x40003b6488, {0x1251d75, 0xb}, 0x400039ae84, 0x0, {0x0, 0x0, 0x0}, ...)
github.com/ellanetworks/core/internal/amf/gmm/message/send.go:305 +0x130
github.com/ellanetworks/core/internal/amf/gmm.HandleMobilityAndPeriodicRegistrationUpdating({0x16870a8, 0x40004872c0}, 0x40003b6488, {0x1251d75, 0xb})
github.com/ellanetworks/core/internal/amf/gmm/handler.go:942 +0x7f4
github.com/ellanetworks/core/internal/amf/gmm.ContextSetup({0x16870a8, 0x40004872c0}, 0x1254ecc?, {0x1252106?, 0xa?}, 0x4000487440)
github.com/ellanetworks/core/internal/amf/gmm/sm.go:405 +0x34c
github.com/ellanetworks/core/internal/util/fsm.(*FSM).SendEvent(0x4000310610, {0x16870a8, 0x40004872c0}, 0x40007a4db0, {0x126658b, 0x14}, 0x4000487440)
github.com/ellanetworks/core/internal/util/fsm/fsm.go:107 +0x1d4
github.com/ellanetworks/core/internal/amf/gmm.HandleSecurityModeComplete({0x16870a8, 0x40004872c0}, 0x40003b6488, {0x1251d75, 0xb}, 0x2e, 0x400050e258)
github.com/ellanetworks/core/internal/amf/gmm/handler.go:2048 +0x458
github.com/ellanetworks/core/internal/amf/gmm.SecurityMode({0x16870a8, 0x40004872c0}, 0x40007a4db0, {0x1252153, 0xb}, 0x40004873b0)
github.com/ellanetworks/core/internal/amf/gmm/sm.go:304 +0xce0
github.com/ellanetworks/core/internal/util/fsm.(*FSM).SendEvent(0x4000310610, {0x16870a8, 0x40004872c0}, 0x40007a4db0, {0x1252153, 0xb}, 0x40004873b0)
github.com/ellanetworks/core/internal/util/fsm/fsm.go:97 +0xf4
github.com/ellanetworks/core/internal/amf/nas.Dispatch({0x16870a8, 0x40004872c0}, 0x40003b6488, {0x1251d75, 0xb}, 0x2e, 0x40004fe9a0)
github.com/ellanetworks/core/internal/amf/nas/dispatch.go:50 +0x6cc
github.com/ellanetworks/core/internal/amf/nas.HandleNAS({0x16870a8, 0x40004872c0}, 0x400030eb40, 0x2e, {0x40004bff90?, 0x0?, 0x0?})
github.com/ellanetworks/core/internal/amf/nas/handler.go:60 +0xec
github.com/ellanetworks/core/internal/amf/ngap.HandleUplinkNasTransport({0x16870a8, 0x40004872c0}, 0x40004c83c0, 0x40000405c0?)
github.com/ellanetworks/core/internal/amf/ngap/handler.go:737 +0x3fc
github.com/ellanetworks/core/internal/amf/ngap.DispatchNgapMsg({0x1690898, 0x400011a460}, 0x40004c83c0, 0x40004d3880)
github.com/ellanetworks/core/internal/amf/ngap/dispatcher.go:321 +0x5c8
github.com/ellanetworks/core/internal/amf/ngap.NgapMsgHandler(...)
github.com/ellanetworks/core/internal/amf/ngap/dispatcher.go:283
github.com/ellanetworks/core/internal/amf/ngap.Dispatch({0x1687070, 0x2512220}, {0x1690898, 0x400011a460}, {0x4000600000, 0x73, 0x20000})
github.com/ellanetworks/core/internal/amf/ngap/dispatcher.go:70 +0x3a4
github.com/ellanetworks/core/internal/amf/ngap/service.handleConnection(0x400011a460, 0x20000, {0x153c388?, 0x153c390?})
github.com/ellanetworks/core/internal/amf/ngap/service/service.go:202 +0x3f8
created by github.com/ellanetworks/core/internal/amf/ngap/service.listenAndServe in goroutine 68
github.com/ellanetworks/core/internal/amf/ngap/service/service.go:139 +0x12f8
```

### Environment

v0.3.1

### Reproduction Steps

N/A

### Expected Behavior

do not crash

### Relevant log output

```shell

```

--- a/internal/amf/ngap/message/forward_ie.go
+++ b/internal/amf/ngap/message/forward_ie.go
@@ -148,7 +148,7 @@ func BuildIEMobilityRestrictionList(ue *context.AmfUe) ngapType.MobilityRestrict
 		}
 	}
 
-	if ue.AmPolicyAssociation.ServAreaRes != nil {
+	if ue.AmPolicyAssociation != nil && ue.AmPolicyAssociation.ServAreaRes != nil {
 		mobilityRestrictionList.ServiceAreaInformation = new(ngapType.ServiceAreaInformation)
 		serviceAreaInformation := mobilityRestrictionList.ServiceAreaInformation
 

